virtual_rainforest/models/animals/dummy_animal_module.py

```diff
@@ -78,6 +82,75 @@ class PalatableSoil:
     """The grid position of the soil pool."""
 
 
+class AnimalCommunity:
+    """This is a class for the animal community of a grid cell."""
+
+    def __init__(
+        self,
+        functional_groups: list[FunctionalGroup],
+    ) -> None:
+        """The constructor of the AnimalCommunity class."""
+        self.functional_groups = functional_groups
+        """A list of all FunctionalGroup types in the model."""
+
+        self.cohorts: dict[str, list[AnimalCohort]] = {
+            k.name: [] for k in self.functional_groups
+        }
+        """Generate a dictionary of functional groups."""
+
+    def immigrate(self, cohort: AnimalCohort, destination: AnimalCommunity) -> None:
+        """Function to move an AnimalCohort between AnimalCommunity objects.
+
+        This function should take a cohort and a destination community and then pop the
+        cohort from this community to the destination.
+
+        Args:
+            cohort: The AnimalCohort moving between AnimalCommunities.
+            destination: The AnimalCommunity the cohort is moving to.
+
+        """
+        immigrant = self.cohorts[cohort.name].pop(
+            self.cohorts[cohort.name].index(cohort)
+        )
+        # Remove target cohort from source community
+        destination.cohorts[immigrant.name].append(immigrant)
+        # Add cohort to destination community
```

Comment:
Personally, I would do this in one go to avoid intermediate variables, but it is totally up to you. 

```suggestion
        destination.cohorts[immigrant.name].append(
            self.cohorts[cohort.name].pop(
                self.cohorts[cohort.name].index(cohort)
            )
        )
```